test/integration/framework/utils.go

```diff
@@ -0,0 +1,358 @@
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/liveness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/readiness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+						},
+					},
+					HostNetwork:  opts.EnableHostNetwork,
+					NodeSelector: nodeSelectors,
+				},
+			},
+		},
+	}
+	return deployment, nil
+}
+
+//TODO: add support to pod-prefixes filtering for logs
+func (o *GardenerTestOperation) fetchPodContainerLogs(podNamespace, podName, containerName string) (*string, error) {
+	logChannel := make(chan string, 1)
+
+	go func() {
+		var (
+			buffer *bytes.Buffer
+			err    error
+		)
+		if len(containerName) == 0 {
+			buffer, err = o.K8sShootClient.GetPodLogs(podNamespace, podName, &corev1.PodLogOptions{})
+		} else {
+			buffer, err = o.K8sShootClient.GetPodLogs(podNamespace, podName, &corev1.PodLogOptions{
+				Container: containerName,
+			})
+		}
+		if err != nil {
+			o.ShootGardenerTest.Logger.Errorf("Something went wrong fetching logs: %s", err.Error())
+			return
+		}
+		logToReturn := buffer.String()
+		logChannel <- logToReturn
+	}()
+
+	select {
+	case result := <-logChannel:
+		return &result, nil
+	case <-time.After(2 * time.Minute):
+		return nil, ErrTimeoutWaitingWaitingForLogs
+	}
+}
+
+// getRunningPodByLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getRunningPodByLabels(labelsMap map[string]string, seedOrShoot string) (*corev1.Pod, error) {
+	var runningPodWithDesiredLabels *corev1.Pod
+	set := labels.Set(labelsMap)
+
+	var podList *corev1.PodList
+	var err error
+
+	switch seedOrShoot {
+	case "seed":
+		podList, err = o.K8sSeedClient.ListPods(o.Shoot.SeedNamespace, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	case "shoot":
+		podList, err = o.K8sShootClient.ListPods(metav1.NamespaceSystem, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		for _, pod := range podList.Items {
+			o.ShootGardenerTest.Logger.Infof("%s:%s", pod.Namespace, pod.Name)
+			if pod.Status.Phase == corev1.PodRunning {
+				runningPodWithDesiredLabels = &pod
+				break
+			}
+		}
+		return runningPodWithDesiredLabels, nil
+	}
+
+	return nil, nil
+}
+
+// getPodByLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getPodByLabels(labelsMap map[string]string, namespace string, seedOrShoot string) (*corev1.Pod, error) {
+	var runningPodWithDesiredLabels *corev1.Pod
```

Comment:
This is duplicate code for the most - split into two methods: Fetch pod depending on your criteria and filter the pod list for sth.

Reply:
Still mostly duplicative.